src/datumaro/components/explorer.py

```diff
@@ -20,9 +20,7 @@ def calculate_hamming(B1, B2):
     :param B2:  vector [r*n]
     :return: hamming distance [r]
     """
-    q = B2.shape[1]  # max inner product value
-    distH = 0.5 * (q - B1 @ B2.transpose())
-    return distH
+    return np.count_nonzero(B1 != B2, axis=1)
```

Comment:
Is there any reason to replace this? I think this could be take more time for bigger matrix.

Reply:
First of all, the previous implement was not computing the hamming distance. It was actually dot product between two vectors in {0,1}^d. Secondly, both (the dot product and this implementation) have the same complexity.